extra-dev/packages/coq-autosubst/coq-autosubst.dev/opam

```diff
@@ -19,7 +19,7 @@ substitutions."""
 build: [make "-j%{jobs}%"]
 install: [make "install"]
 depends: [
-  "coq" {(>= "8.11" & < "8.16~") | (= "dev")}
+  "coq" {(>= "8.14" & < "8.17~") | (= "dev")}
```

Comment:
Maybe we only set the lower bound to avoid failures in CI as Coq versions evolve?
```suggestion
  "coq" {>= "8.14"}
```

Reply:
How do other libraries handle this?

I feel like this should first be done in https://github.com/coq-community/autosubst before we do it here. I am fine leaving the upper version open, but how will we know if/when that breaks?

Reply:
In the `extra-dev` repository, this is quite common. The main reason is that few people use the repository at all, and even fewer make the effort to update packages consistently. Hence, it may be better to skip upper bounds to update packages only when necessary. In contrast, I think `{(>= "8.14" & < "8.17~") | (= "dev")}` is great for in-repo packages, since there, it documents exactly what versions of Coq we are/will test with. And the repo CI is where we will likely detect any breakage.

The cost of using two different idioms is there is some inconsistency in package definitions, but I think it's small enough to justify the gain (of not having to update the `extra-dev` package all the time).

Reply:
It also means that the extra-dev package may break unexpectedly. But I guess that is something people just have to deal with then...

Honestly I don't care very strongly about the autosubst extra-dev package so :shrug: sure, works for me.